### ISSUE TITLE
LR scheduler does not work properly due to AcceleratedOptimizer

### ISSUE THREAD

**Issue by andreatgretel** [microsoft/dp-transformers#43]
Ever since accelerate has been introduced, the optimizer inside the Trainer is doubly wrapped - first DPOptimizer wraps the original Optimizer, and then AcceleratedOptimizer wraps the DPOptimizer. Opacus [does not do the wrapping very carefully](https://github.com/pytorch/opacus/blob/32a465bfa980777e60a763616c83429785aa2ac6/opacus/optimizers/optimizer.py#L243), simply passing `state`, `defaults` and `param_groups` by reference - this reference could easily be replaced, and as a matter of fact accelerate seems to be doing that. In contrast, [the wrapping approach used by the AcceleratedOptimizer](https://github.com/huggingface/accelerate/blob/6af157ea93dfbace1db88b0fdc7dfb568dfdd5a5/src/accelerate/optimizer.py#L85) guarantees that the reference can't be easily replaced.

A consequence of this is that LR schedulers do not seem to work right now. To verify this, you can add the following statements inside the `on_substep_end` callback:
```
print('LR at AcceleratedOptimizer: {}'.format([g['lr'] for g in optimizer.param_groups]))
print('LR at DPOptimizer: {}'.format([g['lr'] for g in dp_optimizer.param_groups]))
print('LR at original Optimizer: {}'.format([g['lr'] for g in dp_optimizer.original_optimizer.param_groups]))
```
The first 2 follow the scheduler, but the 3rd one doesn't, and the LR actually applied once `self.original_optimizer.step()` is called inside of Opacus is precisely that of the original Optimizer.

I only noticed this because I was using a scheduler that started with a LR of 0, and the model parameters weren't changing at all, even if at the 2nd step already LR > 0. It should be harder to notice it in general.

Introducing a new DPOptimizer that inherits from Opacus' but sets param_groups using properties, like the AcceleratedOptimizer does, seems to fix the issue:
```
class DPOptimizer(opacus.optimizers.DPOptimizer):
    @property
    def param_groups(self):
        return self.original_optimizer.param_groups

    @param_groups.setter
    def param_groups(self, param_groups):
        self.original_optimizer.param_groups = param_groups
```

**Comment by wulu473** [comment]
Thanks for looking into this Andre. We should probably report this to opacus directly too.

Thanks for the workaround. If you have a PR ready we can get this fixed quickly, otherwise I can add a fix later this week.

**Comment by andreatgretel** [comment]
Good point about reporting it to Opacus directly, I'll create an issue there too!

I think I won't create the PR, that way you can do the changes however you find better. The fix I did on my local fork is just what I wrote above, nothing else.

Thanks!
